test/steps/suite_steps.go

```diff
@@ -19,14 +19,19 @@ var beforeSuiteMux sync.Mutex
 var gitManager *test.GitManager
 
 // SuiteSteps defines global lifecycle step implementations for Cucumber.
-func SuiteSteps(suite *godog.Suite, fs *FeatureState) {
+func SuiteSteps(suite *godog.Suite, state *ScenarioState) {
 	suite.BeforeScenario(func(scenario *messages.Pickle) {
 		// create a GitEnvironment for the scenario
 		gitEnvironment, err := gitManager.CreateScenarioEnvironment(scenarioName(scenario))
 		if err != nil {
 			log.Fatalf("cannot create environment for scenario %q: %s", scenarioName(scenario), err)
 		}
-		fs.state = scenarioState{gitEnv: gitEnvironment}
+		// Godog only provides state for the entire feature.
+		// We want state to be scenario-specific, hence we reset the shared state before each scenario.
+		// This is a limitation of the current Godog implementation, which doesn't have a `ScenarioContext` method,
+		// only a `FeatureContext` method.
+		// See main_test.go for additional details.
```

Comment:
Is there an issue on godog about a ScenarioContext?

Reply:
Yes, this has been brought up in https://github.com/cucumber/godog/issues/180 and https://github.com/cucumber/godog/issues/75. The consensus seems that the Go implementation should mirror what the other Cucumber implementations do. You know much better than I how they work. I would favor as well if scenarios run concurrently, since all scenarios are independent in Git Town's feature specs. But we have many feature files so aren't affected by this too much except a bit of impedance mismatch from having to re-use a feature state instance for multple scenarios.